pennylane_forest/device.py

```diff
@@ -245,33 +249,34 @@ def apply_rotations(self, rotations):
         """
         rotation_operations = Program()
         for operation in rotations:
-            wires = self.remap_wires(operation.wires)
+            # map the ops' wires to the wire labels used by the device
+            device_wires = self.map_wires(operation.wires)
             par = operation.parameters
-            rotation_operations += self._operation_map[operation.name](*par, *wires)
+            rotation_operations += self._operation_map[operation.name](*par, *device_wires.labels)
 
         return rotation_operations
 
     def reset(self):
         self.prog = Program()
-        self._active_wires = set()
+        self._active_wires = Wires([])
         self._state = None
 
     @property
     def operations(self):
         return set(self._operation_map.keys())
 
-    def mat_vec_product(self, mat, vec, wires):
+    def mat_vec_product(self, mat, vec, device_wire_labels):
```

Comment:
One could also pass the original `wires` object to this method and unpack inside (to follow the guideline of unpacking and translating as late as possible). But here it seemed reasonable to define the function on the device wire labels.

Reply:
makes sense!